src/jukebox/run_jukebox.py

```diff
@@ -1,19 +1,121 @@
+    propagate: no
+  jb.awesome_module:
+    level: DEBUG
+``
+
+.. note::
+The name (and hierarchy path) of the logger can be arbitrary and must not necessarily match the module name (still makes sense)
+There can be multiple loggers per module, e.g. for special classes, to further control the amount of log output
+"""
+
 import os.path
+import sys
 import argparse
+import logging
+import logging.config
+from ruamel.yaml import YAML
 import jukebox.daemon
+import misc.loggingext
 
 
-if __name__ == "__main__":
+def logger_default(level=logging.DEBUG):
+    logger = logging.getLogger('jb')
+    logger.setLevel(level)
+    console = logging.StreamHandler(sys.stdout)
+    console.setLevel(level)
+    formatter = logging.Formatter(
+        fmt='{asctime} - {lineno:4}:{filename:20} - {name} - {threadName:10} - {levelnameColored:8} - '
+            '{lightcyan}{message}{reset}',
+        datefmt='%d.%m.%Y %H:%M:%S',
+        style='{')
+    console.addFilter(misc.loggingext.ColorFilter())
+    console.setFormatter(formatter)
+    logger.addHandler(console)
+    logger.info(f"Enabling default logger with level = {level}")
+    return logger
 
-    # get absolute path of this script
-    dir_path = os.path.dirname(os.path.realpath(__file__))
-    defaultconfigFilePath = os.path.join(dir_path, '../../settings/jukebox.conf')
 
-    argparser = argparse.ArgumentParser(description='The JukeboxDaemon')
-    argparser.add_argument('configuration_file', type=argparse.FileType('r'), nargs='?', default=defaultconfigFilePath)
-    argparser.add_argument('--verbose', '-v', action='count', default=0)
+def logger_configure(filename=None):
+    if filename is None:
+        return logger_default()
+    yaml = YAML(typ='safe')
+    try:
+        with open(filename) as stream:
+            cfg = yaml.load(stream)
+        logging.config.dictConfig(cfg)
+        logger = logging.getLogger('jb')
+    except Exception as e:
+        logger = logger_default()
+        logger.error(f"Using default fallback logger. Reason: while opening '{filename}' for logger configuration")
+        logger.error(f"{e}")
+    # This enforces a fresh file for all RotatingFileHandlers at start of application
+    for h in logger.handlers:
+        if type(h) == logging.handlers.RotatingFileHandler:
+            h.doRollover()
+    return logger
+
 
+def main():
+    # Get absolute path of this script
+    script_path = os.path.abspath(os.path.dirname(os.path.realpath(__file__)))
+    default_cfg_jukebox = os.path.abspath(os.path.join(script_path, '../../settings/jukebox.conf'))
+    default_cfg_logger = os.path.abspath(os.path.join(script_path, '../../settings/logger.yaml'))
+
+    argparser = argparse.ArgumentParser(description='The JukeboxDaemon')
+    argparser.add_argument('-conf', '-c', type=argparse.FileType('r'), default=default_cfg_jukebox,
+                           help=f"jukebox configuration file [default: '{default_cfg_jukebox}'",
+                           metavar="FILE")
```

Comment:
don't mind to switch to named parameter here. Just kept it similar to mpd, 
Didn't think about the system start e.g. via systemd to deep so far. 
I think it will be likely that we don't rely on the default and set it to something during installation. 
But this is not about the -c ;-)

Reply:
Agree. Default is more for development convenience